### PR TITLE
Update tutorial to delete data

### PR DESCRIPTION
**Pull request by surekhasaharan**
Update the deletion tutorial to add the new drop data API from #7439

### CODE REVIEW

**Comment by surekhasaharan** [comment]
Ok, I wasn't sure if those should be removed, will update.

File: docs/content/tutorials/tutorial-delete-data.md [hunk 1]
```diff
@@ -50,27 +50,22 @@ Permanent deletion of a Druid segment has two steps:
 1. The segment must first be marked as "unused". This occurs when a segment is dropped by retention rules, and when a user manually disables a segment through the Coordinator API. This tutorial will cover both cases.
```
Comment (gianm) [inline docs/content/tutorials/tutorial-delete-data.md]: Please remove the mention of 'retention rules' here, as well as the reference to "Tutorial: Configuring retention" being 'highly recommended' above. (We shouldn't be using retention rules anymore after these edits)

Comment (surekhasaharan) [reply docs/content/tutorials/tutorial-delete-data.md]: sure, removed all references to 'retention' rules in this tutorial.

**Comment by fjy** [comment]
👍

**Pull request merged**